yabt/graph_test.py

```diff
@@ -207,61 +213,81 @@ def test_target_graph_intenral_dir(basic_conf):
         set(build_context.target_graph.edges))
 
 
-def test_stable_topological_sort():
-    """Test that my modified topological sort is stable.
-
-    Note - not doing many cycles because I saw that even with the non-stable
-    implementation, it is stable in the context of the same process...
-    (only reruns showed the unstable behavior)
+def subgraph_stable_topsort_test(graph):
+    """Test that sort is stable for subgraph"""
+    def shuffle_graph(graph):
+        l_g = list(graph.nodes())
+        random.shuffle(l_g)
+        nx_g = networkx.DiGraph()
+        for name in l_g:
+            nx_g.add_node(name)
+        l_e = list(graph.edges())
+        for e_d in l_e:
+            nx_g.add_edge(e_d[0], e_d[1])
+        return nx_g
+    top_sort_l = list(topological_sort(graph))
+    hash_res0 = dict()
+    for root in get_graph_roots(graph):
+        hash_res0[root] = tu.calc_node_hash(graph, top_sort_l, root)
+    for i in range(5):
+        cur_g = shuffle_graph(graph)
+        top_sort_l = list(topological_sort(cur_g))
+        for root in get_graph_roots(graph):
+            hash = tu.calc_node_hash(cur_g, top_sort_l, root)
+            assert hash_res0[root] == hash
+    for root in get_graph_roots(graph):
```

Comment:
Why do you need this too? isn't the previous for enough for the test? how is it different?